packages/fxa-admin-panel/src/components/AccountSearch/index.tsx

```diff
@@ -49,20 +49,82 @@ export const GET_ACCOUNT_BY_EMAIL = gql`
   }
 `;
 
+export const GET_EMAILS = gql`
+  query fetchAll($search: String!) {
+    getEmailsLike(search: $search) {
+      email
+    }
+  }
+`;
+
 export const AccountSearch = () => {
   const [inputValue, setInputValue] = useState<string>('');
   const [showResult, setShowResult] = useState<Boolean>(false);
+  const [suggestions, setSuggestion] = useState<Array<string>>([]);
+  const [text, setText] = useState<string>('');
   const [getAccount, { loading, error, data, refetch }] = useLazyQuery(
     GET_ACCOUNT_BY_EMAIL
   );
+  const [getEmailLike, { data: data1, refetch: refetch1 }] = useLazyQuery(
+    GET_EMAILS
+  );
 
   const handleSubmit = (event: React.FormEvent) => {
     event.preventDefault();
-    getAccount({ variables: { email: inputValue } });
+    getAccount({ variables: { email: text } });
     // Don't hide after the first result is shown
     setShowResult(true);
   };
 
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setSuggestion([]);
+    onTextChanged(event);
+    getEmailLike({ variables: { search: event.target.value } });
+    //getEmails();
+  };
+
+  const onTextChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    let filtered_list = [];
+    if (value.length > 0) {
+      console.log('value:', value);
+      getEmailLike({ variables: { search: value } });
+
+      if (data1 != null) {
+        console.log(data1);
+        for (let i = 0; i < data1.getEmailsLike.length; i++) {
+          filtered_list[i] = data1.getEmailsLike[i].email;
+        }
+      }
+    }
+    const regex = new RegExp(`^${value}`, 'i');
+    filtered_list = filtered_list.sort().filter((v) => regex.test(v));
```

Comment:
If we change our query on the server side we won't need to filter these results and the sql query will be more efficient. See my comment in `account.resolver.ts`